### PR TITLE
Rework chat.muc.join

### PR DESCRIPTION
**Pull request by dimaspirit**
**Made/Proposed changes:**
- Rework muc.join callback's parameters; Now uses Node.js callbacks approach instead of returned stanza.
```
QB.chat.muc.join(dialogId, function(error, response) {
   if(error) {
      console.log('Error is null when all is Ok');
  }

  console.log(`response.dialogId` is always contains in response);
});
```
- Rework putting parameters of muc.join. Now could pass jid or id of dialog.

**How should this be manually tested?**
Run jasmine

**Does the documentation need an update?**
Yes

### CODE REVIEW

File: src/modules/chat/qbChat.js [hunk 1]
```diff
@@ -2354,6 +2388,18 @@ Helpers.prototype = {
...
         this._userCurrentJid = jid;
+    },
+
+    getDialogJid: function(identifier) {
+        var jid;
+
+        if(identifier.indexOf('@') > 0) {
+            jid = identifier;
+        } else {
+            jid = config.creds.appId + '_' + identifier + '@' + config.endpoints.muc;
```
Comment (soulfly) [inline src/modules/chat/qbChat.js]: We have a helper for this, please use it

**Comment by soulfly** [comment]
It looks like it's now backward incompatible change with muc join callback.

I propose to save backward compatibility. If a user pass only 1 argument in callback  - pass 1 argument as previous. If 2 -   then error+response.

**Review by Vladlukhanin — APPROVED** [review]

File: src/modules/chat/qbChat.js [hunk 2]
```diff
@@ -1495,19 +1495,21 @@ MucProxy.prototype = {
     /**
      * Join to the group dialog. {@link https://quickblox.com/developers/Web_XMPP_Chat_Sample#Chat_in_group_dialog More info.}
      * @memberof QB.chat.muc
-     * @param {String} dialogJid - Use dialog jid to join to this dialog.
+     * @param {String} dialogIdentifier - Use dialog jid or dialog id to join to this dialog.
      * @param {joinMacCallback} callback - The callback function.
      * */
-    join: function(dialogJid, callback) {
+    join: function(dialogIdentifier, callback) {
```
Comment (soulfly) [inline src/modules/chat/qbChat.js]: let's name it dialogIdOrJid

**Comment by soulfly** [comment]
Approved

**Pull request merged**